magma/ref.py

```diff
@@ -196,8 +196,9 @@ def set_defn(self, defn):
 
 
 class DerivedRef(Ref):
-    def __init__(self, parent):
+    def __init__(self, parent, index):
         self._parent = parent
+        self.index = index
```

Comment:
let's make this private and use an accessor function for uniformity: `def index(self): return self._index`

(property is probably better but just for uniformity's sake)

Reply:
wait is this needed for this change to work?

Reply:
Yes, it's needed to generically refer to the index of a reference without knowing whether it's an array or tuple (both derivedref's have an index